src/Analysis.cpp

```diff
@@ -374,380 +237,59 @@ void populateAliasedVars(Instruction* I, SetVector<Instruction *>& workSet, Alia
-          true
-        }; 
-        
-      
-      }
-
-    
-      // fill in the rest 
-      for (auto Pair1 : PriorPostCM) {
-        std::string varName = Pair1.first; 
-        CM priorPostCM = Pair1.second; 
-
-        if (!lub[varName].setInitialized) {
-          
-          lub[varName] = {
-            std::set<std::string>(priorPostCM.cmSet), 
-            true 
-          }; 
-
-        }
-      }
-
-      PreCalledMethods[currentBranch] = std::map<std::string, CM>(lub);
-
-      for (Instruction* I : instructions) {
-        transfer(I, instructions, lub, AliasedVars);
-      }
-
-      PostCalledMethods[currentBranch] = lub; 
-
-      for (CFG* succ : cfg->getSuccessors()) {
-        logout("analyzing succesor " << succ->getBranchName())
-        analyzeCFG(succ, PreCalledMethods, PostCalledMethods, AliasedVars, currentBranch); 
-      }
-      
-
-    }
-    else {
-      logout("doing normal flow for " << currentBranch << " and prior = " << priorBranch)  
-
-      std::map<std::string, CM> priorPostCM = std::map<std::string, CM>(PostCalledMethods[priorBranch]); 
-
-      PreCalledMethods[currentBranch] = std::map<std::string, CM>(priorPostCM); 
-      llvm::SetVector<Instruction*> instructions = cfg->getInstructions(); 
-
-      std::map<std::string, CM> flowInto = std::map<std::string, CM>(PostCalledMethods[priorBranch]); 
-
-       for (Instruction* I : instructions) {
-        transfer(I, instructions, flowInto, AliasedVars);
-      }
-
-      PostCalledMethods[currentBranch] = flowInto;
-
-      
-
-      for (CFG* succ : cfg->getSuccessors()) {
-        analyzeCFG(succ, PreCalledMethods, PostCalledMethods, AliasedVars, currentBranch); 
-      }
-
-    }
-
-  }
-
-}
 
 
 void CalledMethodsAnalysis::doAnalysis(Function &F, PointerAnalysis *PA, std::string optLoadFileName) {
   SetVector<Instruction *> WorkSet;
   SetVector<Value *> PointerSet;
   std::string fnName = F.getName().str(); 
 
-  int slashCount = 0; 
   std::string testName = getTestName(optLoadFileName); 
 
   bool functionIsKnown = false; 
   logout("fnname = " << fnName << " opt load file name = " << testName)
 
-  loadFunctions();
-  loadExpectedResult(testName, ExpectedResult);
- 
+  if (!loadAndBuild) {
+    loadFunctions();
+    calledMethods.buildExpectedResult(testName);
+    mustCall.buildExpectedResult(testName);
+    ALLOW_REDEFINE = getAllowedRedefine(testName); 
+    loadAndBuild = true; 
+  }
+
 
   if (!ALLOW_REDEFINE) {
         // check if code re-defines a safe function or memory function. if so, cast it as a unsafe function
     if (SafeFunctions[fnName]) {
       SafeFunctions[fnName] = false; 
       UnsafeFunctions[fnName] = true; 
       functionIsKnown = true; 
-      errs() << "CME-WARNING: Re-definition of safe function '" << fnName << "' identified on  and will be labelled as unsafe.\n";
+      errs() << "**ANALYSIS-WARNING**: Re-definition of safe function '" << fnName << "' identified on  and will be labelled as unsafe.\n";
```

Comment:
I think there is something missing from this error message (after "on")

Reply:
Yeah this should just say "identifed and will be ..."